kyo-scheduler/jvm/src/test/scala/kyo/scheduler/SchedulerTest.scala

```diff
@@ -123,6 +123,19 @@ class SchedulerTest extends AnyFreeSpec with NonImplicitAssertions {
         }
     }
 
+    "dispatches java interrupt" in withScheduler { scheduler =>
+        val cdl = new CountDownLatch(1)
+        scheduler.schedule(TestTask(_run = () => {
```

Comment:
I am concerned this may dispatch an interrupt to a blocked, but working task. Can we add a test for computing something blocking, eg md5 hashes for some time period greater than the default java interrupt time?

Reply:
Yes, it'll first try preempting the task and then dispatching the thread interrupt if the preemption didn't work. I think we could dispatch the thread interrupt only if it's blocked. Detecting blocking isn't fail-proof, though. There are cases where the blocking happens in native code, which is not possible to detect by inspecting the thread's state.